Tests/TestingTests/TypeInfoTests.swift

```diff
@@ -50,6 +50,56 @@ struct TypeInfoTests {
     #expect(TypeInfo(describing: T.self).fullyQualifiedName == "(Swift.Int, Swift.String) -> Swift.Bool")
   }
 
+  @Test("Splitting raw identifiers",
+    arguments: [
+      ("Foo.Bar", ["Foo", "Bar"]),
+      ("`Foo`.Bar", ["`Foo`", "Bar"]),
+      ("`Foo`.`Bar`", ["`Foo`", "`Bar`"]),
+      ("Foo.`Bar`", ["Foo", "`Bar`"]),
+      ("Foo.`Bar`.Quux", ["Foo", "`Bar`", "Quux"]),
+      ("Foo.`B.ar`.Quux", ["Foo", "`B.ar`", "Quux"]),
+
+      // These have substrings we intentionally strip out.
+      ("Foo.`B.ar`.(unknown context at $0).Quux", ["Foo", "`B.ar`", "Quux"]),
+      ("(extension in Module):Foo.`B.ar`.(unknown context at $0).Quux", ["Foo", "`B.ar`", "Quux"]),
+      ("(extension in `Module`):Foo.`B.ar`.(unknown context at $0).Quux", ["Foo", "`B.ar`", "Quux"]),
+      ("(extension in `Module`):`Foo`.`B.ar`.(unknown context at $0).Quux", ["`Foo`", "`B.ar`", "Quux"]),
```

Comment:
In addition to testing the issue with the colon above, another good test case would be one where someone named their module `(extension in Foo):Bar`; i.e., ``"`(extension in Foo):Bar`.Baz"`` -> ``["`(extension in Foo):Bar`", "Baz"]``.

And then an extension on top of that one with some nesting, because why not: ``(extension in `(extension in Foo2):Bar2`):`(extension in Foo):Bar`.Baz`` 😵‍💫

Reply:
Congrats, @allevato, you invented recursion! Now get `NOT` into symbol naming somehow and it'll be Turing-complete!